### PR TITLE
Fix Iterable Index error

### PR DESCRIPTION
**Pull request by franz101**
- Quickest fix in history of DQ
Turns out the error was caused because the indices were a generator not a list.
Added test

### CODE REVIEW

File: dataquality/integrations/fastai.py [hunk 1]
```diff
@@ -283,6 +278,11 @@ def after_pred(self) -> None:
         # Store the current batch ids by trimming the stored ids by
         # the batch size length
         cur_split = self.logger_config.cur_split
+        indices = self.idx_store[FAIKey.dataloader_indices][cur_split][-1]
+        # check if list or iterator and convert to list
+        if not isinstance(indices, list):
+            self.idx_store[FAIKey.dataloader_indices][cur_split][-1] = indices
```
Comment (Ben-Epstein) [inline dataquality/integrations/fastai.py:284]: shouldnt this be
```suggestion
            self.idx_store[FAIKey.dataloader_indices][cur_split][-1] = list(indices)
```

File: tests/integrations/fastai/test_text_fai.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    mock_get_project_by_name: MagicMock,
+    set_test_config: Callable,
+    mock_wait_for_run: MagicMock,
+    mock_make_request: MagicMock,
+    mock_upload_log_file: MagicMock,
+    mock_reset_run: MagicMock,
+    mock_version_check: MagicMock,
+    cleanup_after_use: Generator,
+) -> None:
+    global df
```
Comment (setu4993) [inline tests/integrations/fastai/test_text_fai.py:66]: My 0.02: We should avoid using global dfs for tests.

File: tests/integrations/fastai/test_text_fai.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    mock_get_project_by_name.return_value = {"id": DEFAULT_PROJECT_ID}
+    mock_create_run.return_value = {"id": DEFAULT_RUN_ID}
+    set_test_config(current_project_id=None, current_run_id=None)
+    dls = TextDataLoaders.from_df(
+        df, text_col="text", label_col="label", drop_last=False, bs=2
+    )
+    dq.init(task_type=TaskType.text_classification)
+    labels = dls.vocab[-1]
+    print("labels")
+    print(labels)
```
Comment (setu4993) [inline tests/integrations/fastai/test_text_fai.py:76]: I don't think we want prints in tests?
```suggestion
```

**Review by setu4993 — APPROVED** [review]
Left 1 note about using a global DF, but rest seems fine to me.

File: tests/integrations/fastai/test_text_fai.py (hunk 2 above)
Comment (franz101) [reply tests/integrations/fastai/test_text_fai.py:66]: sure thing

**Pull request merged**